pipeline.cpp

```diff
@@ -71,6 +76,39 @@ void DrawPoint(Buffer2D<PIXEL> & target, Vertex* v, Attributes* attrs, Attribute
 void DrawLine(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* const attrs, Attributes* const uniforms, FragmentShader* const frag)
 {
     // Your code goes here
+    for (float t = 0; t < 1; t += .01)
+    {
+        int x = ((int)triangle[0].x)*(1.0-t) + ((int)triangle[1].x)*t;
+        int y = ((int)triangle[0].y)*(1.0-t) + ((int)triangle[1].y)*t;
+        target[y][x] = attrs[0].color;
+    }
+
+    for (float t = 1; t < 2; t += .01)
+    {
+        int x = ((int)triangle[1].x)*(1.0-t) + ((int)triangle[2].x)*t;
+        int y = ((int)triangle[1].y)*(1.0-t) + ((int)triangle[2].y)*t;
+        target[y][x] = attrs[1].color;
+    }
+
+    for (float t = 2; t < 3; t += .01)
+    {
+        int x = ((int)triangle[2].x)*(1.0-t) + ((int)triangle[0].x)*t;
+        int y = ((int)triangle[2].y)*(1.0-t) + ((int)triangle[0].y)*t;
+        target[y][x] = attrs[2].color;
+    }
+
+}
+
+/*******************************************************
+ * crossProduct
+ *   By taking the two vertex point we can times them together and
+ *   Perform the cross product of the two and return the float
+ * ******************************************************/
+float crossProduct(Vertex one, Vertex two)
+{
+    float crossPro = 0.0;
```

Comment:
Good job in declaring a float before assigning it.  I see this helps in ensuring what return value will result from the operation.

I think the Vertex class uses Doubles for its attributes.  If this is the case, there is the possibility of data loss going from a double to a float.